lib/plugin_helper.rb

```diff
@@ -0,0 +1,66 @@
+module RubyPluginHelper
+  # 'template' is the value of `target_mapping` from the inventory
+  # Returns an array of strings which are the lookups from the template
+  def config_lookup(template)
+    lookups = []
+    postwalk_vals(template) do |value|
+      # First collect the lookups to make
+      lookups << value if value.is_a?(String)
+      value
+    end
+    lookups.uniq.map { |lookup| lookup.split('.') }
+  end
+
+  # 'template' is the value of `target_mapping` from the inventory
+  # 'lookups' is an array of hashes, where each hash is the lookup for a target
+  # Returns an array of hashes to be sent to the inventory
+  def populate_mapping(template, lookups)
+    lookups.map do |lookup|
+      postwalk_vals(template) do |value|
+        if value.is_a?(String)
+          segments = value.split('.').map do |segment|
+            begin
+              # Turn it into an integer if we can
+              Integer(segment)
+            rescue ArgumentError
+              # Otherwise return the value
+              segment
+            end
+          end
+          answer = lookup.dig(*segments)
+          warn("Could not resolve lookup #{value} with #{lookup}") unless answer
```

Comment:
How is this surfaced? Will the task helper collect in on stderror? Or does it just write to stderr no matter what?

Reply:
Hm it doesn't. I'm inclined to just let this be nil, since I think there are valid cases where a lookup can return nil rather than raise an error. I'm not sure there's a better way to log this....